pkg/auth/authorizer/abac/abac.go

```diff
@@ -98,7 +99,15 @@ func NewFromFile(path string) (policyList, error) {
 }
 
 func (p policy) matches(a authorizer.Attributes) bool {
-	if p.subjectMatches(a) {
+
+	switch {
+	case p.NonResourcePath != "":
```

Comment:
`subjectMatches` and readonly checks still apply, move the nonResourcePath check to be a peer to the resource checking